pkg/metadata/metadata.go

```diff
@@ -111,6 +111,32 @@ func HasConfigSyncMetadata(obj client.Object) bool {
 	return false
 }
 
+// ConfigSyncMetadata contains fields needed to set all Config Sync metadata on
+// a managed resource.
+type ConfigSyncMetadata struct {
+	ApplySetID      string
```

Comment:
linter was happy with these public fields not having descriptions?

Reply:
Evidently so. Added comments